app/src/main-process/shell.ts

```diff
@@ -0,0 +1,25 @@
+import * as Url from 'url'
+import { shell } from 'electron'
+
+/**
+ * Wraps the inbuilt shell.openItem path to address a focus issue that affects macOS.
+ *
+ * When opening a folder in Finder, the window will appear behind the application
+ * window, which may confuse users. As a workaround, we will fallback to using
+ * shell.openExternal for macOS until it can be fixed upstream.
+ *
+ * @param path directory to open
+ */
+export function openPathSafe(path: string) {
```

Comment:
Can we call this `openDirectorySafe` _or_ automatically detect whether it's a directory or not and use `shell.showItemInFolder` if it's not? It feels a bit dangerous to me to call a method 'Safe' when, if passed a path to an executable we'd end up executing it.